### DISCUSSION TITLE
Bad performance when sending more than 100 lines

### DISCUSSION THREAD

**Discussion by mkroman** [open-webui/open-webui#7986] [Issues]
# Bug Report

When I try to start a new chat containing more than a couple of hundreds of lines, the browser freezes for several seconds before actually sending the request to the backend.

## Installation Method

I'm running Open WebUI in a Docker container, with host network access, which is where ollama is running.

## Environment

- **Open WebUI Version:** v0.4.8
- **Ollama (if applicable):** v0.5.2
- **Operating System:** Arch Linux
- **Browser (if applicable):** Firefox 133.0.3

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

The experience should be smooth and the request should be sent immediately.

## Actual Behavior:

The browser hangs for several seconds before sending the request.

## Description

**Bug Summary:**

When I try to start a new chat containing more than a couple of hundreds of lines, the browser freezes for several seconds before actually sending the request to the backend.

## Reproduction Details

**Steps to Reproduce:**

1. Set up openwebui
2. Add a large context model like Gemini Pro 1.5
3. Fill the text field with the contents of `prompt.txt`
4. The browser should hang

## Additional Information

I profiled the process and the result can be viewed [here](https://share.firefox.dev/4gwwltl).

[prompt.txt](https://github.com/user-attachments/files/18216775/prompt.txt)

**Comment by tjbck** [comment]
If you could share a convo export that would be helpful here!

**Reply by mkroman** [reply]
I don't think the export adds anything of value as the problem seems to exist somewhere in the frontend, in the editor, just before submitting the message.

Not sure what format you'd like, but here's a JSON export.

[chat-export-1734764857687.json](https://github.com/user-attachments/files/18217644/chat-export-1734764857687.json)

**Comment by crizCraig** [comment]
I'm also experiencing slowness, especially typing latency (like 1 character per second) with large prompts like this 

https://gist.github.com/crizCraig/cf7432465413e820b2bebc6c85449c33

Perhaps there are onkeyup processors or other preprocessing going on which for large text we may want to disable 🤷 

Thanks @tjbck for your continued awesome work on OI!!!
